api/resource/factory.go

```diff
@@ -213,7 +202,10 @@ func (rf *Factory) inlineAnyEmbeddedLists(
 		}
 		items, ok := m["items"]
```

Comment:
I worry about what happens when an empty PodList is put.

```
apiVersion: v1
kind: PodList
items: []
```

Can you add that test case?